repo/repo

```diff
@@ -286,6 +285,12 @@ function urldecode() {
     printf '%b' "${1//%/\\x}"
 }
 
+function jcr_to_filesystem() {
+    local jcrPath=$1
+    # Handle namespaces and a default empty one
+    echo ${jcrPath} | sed -e 's#\/\([^:]*\):#/_\1_#g' | sed -e 's#/:#/#g'
```

Comment:
@alexkli: Thanks for thorough review! It makes sense to reiterate on this and fix/simplify the code. At least it was working for me. Let me find some time this week to fix this and I hope we can finish it soon.

My replies (but not code yet fixed):

1. Right, regexp usually takes the longest string that is able to match so we need to include both boundaries: `/` & `:` instead of just `:`
2. `/:` is a special allowed case of a default namespace (I briefly reviewed JCR standard looking for so called XML namespace prefix). The problem I see here is that we will end up with: `__propertyName` which is not exactly the same as `propertyName`. We probably should review how Vault deals with such things under the hood. I'm okay to drop this if Vault manages double `_`.
3. Probably because of `_` escape `__` will be treated as `_` in that case so we cannot drop `2.`